### ISSUE TITLE
Handle AttributeError in DataSet REST API

### ISSUE THREAD

**Issue by hackdna** [refinery-platform/refinery-platform#1734]
* Specific code commit: 7367df5
* Version of the web browser and OS: N/A
* Environment where the error occurred: AWS instance

### Steps to reproduce
Unknown: traceback received by email, possibly as a result of a data set import

### Observed behavior
```
Internal Server Error: /api/v1/data_sets/
Traceback (most recent call last):
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/tastypie/resources.py", line 202, in wrapper
    response = callback(request, *args, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/tastypie/resources.py", line 433, in dispatch_list
    return self.dispatch('list', request, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/tastypie/resources.py", line 465, in dispatch
    response = method(request, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/tastypie/resources.py", line 1298, in get_list
    bundles.append(self.full_dehydrate(bundle, for_list=True))
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/tastypie/resources.py", line 861, in full_dehydrate
    bundle = self.dehydrate(bundle)
  File "/srv/refinery-platform/refinery/core/api.py", line 541, in dehydrate
    bundle.data["version"] = bundle.obj.get_version_details().version
AttributeError: 'NoneType' object has no attribute 'version'

Request repr():
<WSGIRequest
path:/api/v1/data_sets/,
GET:<QueryDict: {u'order_by': [u'-modification_date'], u'offset': [u'50'], u'limit': [u'50'], u'format': [u'json']}>,
POST:<QueryDict: {}>,
COOKIES:{'_ga': 'GA1.2.2126546047.1492614465',
'_gat': '1',
'_gid': 'GA1.2.844249910.1494882001',
'csrftoken': <redacted>,
'sessionid': <redacted>},
META:{'CONTEXT_DOCUMENT_ROOT': '/var/www',
'CONTEXT_PREFIX': '',
u'CSRF_COOKIE': <redacted>,
'DOCUMENT_ROOT': '/var/www',
'GATEWAY_INTERFACE': 'CGI/1.1',
'HTTP_ACCEPT': 'application/json, text/plain, */*',
'HTTP_ACCEPT_ENCODING': 'gzip, deflate, sdch',
'HTTP_ACCEPT_LANGUAGE': 'en-US,en;q=0.8,pl;q=0.6',
'HTTP_CONNECTION': 'keep-alive',
'HTTP_COOKIE': 'sessionid=<redacted>; csrftoken=<redacted>; _ga=GA1.2.2126546047.1492614465; _gid=GA1.2.844249910.1494882001; _gat=1',
'HTTP_DNT': '1',
'HTTP_HOST': 'beta.stemcellcommons.org',
'HTTP_REFERER': 'http://beta.stemcellcommons.org/',
'HTTP_USER_AGENT': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.36',
'HTTP_X_CSRFTOKEN': <redacted>,
'HTTP_X_FORWARDED_FOR': '134.174.140.132',
'HTTP_X_FORWARDED_PORT': '80',
'HTTP_X_FORWARDED_PROTO': 'http',
'PATH_INFO': u'/api/v1/data_sets/',
'PATH_TRANSLATED': '/srv/refinery-platform/refinery/config/wsgi_aws.py/api/v1/data_sets/',
'QUERY_STRING': 'format=json&limit=50&offset=50&order_by=-modification_date',
'REMOTE_ADDR': '172.31.13.218',
'REMOTE_PORT': '17218',
'REQUEST_METHOD': 'GET',
'REQUEST_SCHEME': 'http',
'REQUEST_URI': '/api/v1/data_sets/?format=json&limit=50&offset=50&order_by=-modification_date',
'SCRIPT_FILENAME': '/srv/refinery-platform/refinery/config/wsgi_aws.py',
'SCRIPT_NAME': u'',
'SERVER_ADDR': '172.31.2.24',
'SERVER_ADMIN': '[no address given]',
'SERVER_NAME': 'beta.stemcellcommons.org',
'SERVER_PORT': '80',
'SERVER_PROTOCOL': 'HTTP/1.1',
'SERVER_SIGNATURE': '',
'SERVER_SOFTWARE': 'Apache/2.4.7 (Ubuntu)',
'mod_wsgi.application_group': 'beta.stemcellcommons.org|',
'mod_wsgi.callable_object': 'application',
'mod_wsgi.enable_sendfile': '0',
'mod_wsgi.handler_script': '',
'mod_wsgi.input_chunked': '0',
'mod_wsgi.listener_host': '',
'mod_wsgi.listener_port': '80',
'mod_wsgi.process_group': 'refinery',
'mod_wsgi.queue_start': '1494882085447419',
'mod_wsgi.request_handler': 'wsgi-script',
'mod_wsgi.script_reloading': '1',
'mod_wsgi.version': (3, 4),
'wsgi.errors': <mod_wsgi.Log object at 0x7f88bc2dfcb0>,
'wsgi.file_wrapper': <built-in method file_wrapper of mod_wsgi.Adapter object at 0x7f88bc968eb8>,
'wsgi.input': <mod_wsgi.Input object at 0x7f88bc2dfdf0>,
'wsgi.multiprocess': False,
'wsgi.multithread': True,
'wsgi.run_once': False,
'wsgi.url_scheme': 'http',
'wsgi.version': (1, 0)}>
```

### Expected behavior
No error.

**Comment by scottx611x** [comment]
I think this scenario occurs if someone refeshes the Main page, or hits the `/api/v1/datasets` in the middle of a DataSet's creation. Could be helpful for reproducing.

**Comment by scottx611x** [comment]
Addressed by #1739
